lib/org/chromium/webidl/HTMLFileContent.js

```diff
@@ -0,0 +1,94 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+'use strict';
+
+foam.CLASS({
+  package: 'org.chromium.webidl',
+  name: 'HTMLFileContent',
+
+  requires: [
+    'foam.parsers.HTMLLexer'
+  ],
+
+  properties: [
+    {
+      name: 'file',
+      // description: 'content of the HTML file',
+      postSet: function(_, file) {
+        this.pre = this.parse(file);
+      }
+    },
+    {
+      name: 'pre',
+      // description: 'all pre tags found in the parsed HTML file',
+    }
+  ],
+
+  methods: [
+    function parse(str) {
+      var self = this;
+      var lexer = self.HTMLLexer.create();
+      var OPEN = lexer.TagType.OPEN.name;
+      var CLOSE = lexer.TagType.CLOSE.name;
+      var OPEN_CLOSE = lexer.TagType.OPEN_CLOSE.name;
+
+      var results = lexer.parseString(str).value;
+
+      // Tag matching
+      // Using a stack to obtain ancestorOf <pre class=idl> tags
+      var tags = [];
+      var pre = [];
+      results.forEach(function(node, i) {
+        var item = results[i];
+
+        if (foam.core.FObject.isInstance(item)) {
+          var top = tags[tags.length - 1];
+          if (top === undefined || item.type.name === OPEN) {
```

Comment:
I don't think you need the top defined check. Only open tags should be pushed, so if you somehow ended up with an empty stack and either an extra close tag (or plain text, if parseString returns text), not pushing is probably the right thing to do.